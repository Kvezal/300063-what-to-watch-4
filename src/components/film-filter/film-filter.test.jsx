import React from "react";
import render from "react-test-renderer";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import FilmFilter from "./film-filter";


Enzyme.configure({
  adapter: new Adapter(),
});

describe(`FilmFilterComponent`, () => {
  const filterList = [
    {name: `Documentary`, value: `Documentary`},
    {name: `Dramas`, value: `Drama`},
    {name: `Horror`, value: `Horror`}
  ];

  test(`should render component`, () => {
    const tree = render
      .create(
          <FilmFilter
            list={filterList}
            onItemClick={() => {}}
            activeItem={filterList[0].value}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should create component`, () => {
    const FilmFilterComponent = shallow(
        <FilmFilter
          list={filterList}
          onItemClick={() => {}}
          activeItem=""
        />
    );
    const filmFilter = FilmFilterComponent.find(`ul.catalog__genres-list`);
    expect(filmFilter).toHaveLength(1);
  });

  test(`should have filter list`, () => {
    const FilmFilterComponent = shallow(
        <FilmFilter
          list={filterList}
          onItemClick={() => {}}
          activeItem=""
        />
    );
    const filterItems = FilmFilterComponent.find(`li.catalog__genres-item`);
    expect(filterItems).toHaveLength(filterList.length);
  });

  test(`filter item should be pressed`, () => {
    const onItemClick = jest.fn();
    const FilmFilterComponent = shallow(
        <FilmFilter
          list={filterList}
          onItemClick={onItemClick}
          activeItem=""
        />
    );
    const filterItems = FilmFilterComponent.find(`li.catalog__genres-item`);
    filterItems.forEach((item) => item.simulate(`click`));
    expect(onItemClick).toBeCalledTimes(filterItems.length);
  });

  test(`filter item should return value after click`, () => {
    const onItemClick = jest.fn();
    const FilmFilterComponent = shallow(
        <FilmFilter
          list={filterList}
          onItemClick={onItemClick}
          activeItem=""
        />
    );
    const filterItems = FilmFilterComponent.find(`li.catalog__genres-item`);
    filterList.forEach((filter, index) => {
      filterItems.at(index).simulate(`click`);
      expect(onItemClick).toBeCalledWith(filter.value);
    });
  });

  test(`should have one active item`, () => {
    const FilmFilterComponent = shallow(
        <FilmFilter
          list={filterList}
          onItemClick={() => {}}
          activeItem={filterList[0].value}
        />
    );
    const filmFilter = FilmFilterComponent.find(`li.catalog__genres-item.catalog__genres-item--active`);
    expect(filmFilter).toHaveLength(1);
  });
});
