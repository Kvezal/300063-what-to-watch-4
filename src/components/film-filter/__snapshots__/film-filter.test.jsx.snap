// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmFilterComponent should render component 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      Horror
    </a>
  </li>
</ul>
`;
